core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSailValidationReportHelper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * *****************************************************************************
+ * Copyright (c) 2015 Eclipse RDF4J contributors, Aduna, and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ * *****************************************************************************
+ */
+
+package org.eclipse.rdf4j.sail.shacl;
+
+import java.lang.invoke.MethodHandles;
+
+import org.eclipse.rdf4j.common.exception.ValidationException;
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.Rio;
+import org.eclipse.rdf4j.rio.WriterConfig;
+import org.eclipse.rdf4j.rio.helpers.BasicWriterSettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @since 4.0.0
+ * @author Florian Kleedorfer
+ */
+public class ShaclSailValidationReportHelper {
+	private static final WriterConfig WRITER_CONFIG = new WriterConfig();
+
+	static {
+		WRITER_CONFIG
+				.set(BasicWriterSettings.PRETTY_PRINT, true)
+				.set(BasicWriterSettings.INLINE_BLANK_NODES, true);
+	}
+
+	public static void logValidationReportFromThrowableCause(Throwable t) {
+		Throwable thowable;
+		for (thowable = t; thowable != null; thowable = thowable.getCause()) {
+			if (thowable instanceof ValidationException) {
+				break;
+			}
+		}
+		if (thowable != null && thowable instanceof ValidationException) {
+			Model model = ((ValidationException) thowable).validationReportAsModel();
+			Rio.write(model, System.err, RDFFormat.TURTLE, WRITER_CONFIG);
+		}
+	}
```

Comment:
Could we pass in the OutputStream instead of hard coding it, especially since the method name says 'log' but there isn't a logger.

Reply:
I don't like that System.err is hard-coded here either. I actually would want the output to be end up in the logger, but using the logger directly would require to pass the loglevel (or hardcode that). I'll make the change you suggest and add a method that just returns the output as a string, so it's easier to log it if that's what you want